File: templates/Features/StoreNotifications/Services/StoreNotificationsFeatureService.cs
using Microsoft.Services.Store.Engagement;

using System;
using System.Threading.Tasks;

using Windows.ApplicationModel.Activation;

using Param_RootNamespace.Activation;

namespace Param_RootNamespace.Services
{
    internal class StoreNotificationsFeatureService : ActivationHandler<ToastNotificationActivatedEventArgs>
    {
        public async Task InitializeAsync()
        {
            StoreServicesEngagementManager engagementManager = StoreServicesEngagementManager.GetDefault();
            await engagementManager.RegisterNotificationChannelAsync();
        }

        protected override async Task HandleInternalAsync(ToastNotificationActivatedEventArgs args)
        {
            var toastActivationArgs = args as ToastNotificationActivatedEventArgs;

            StoreServicesEngagementManager engagementManager = StoreServicesEngagementManager.GetDefault();
            string originalArgs = engagementManager.ParseArgumentsAndTrackAppLaunch(toastActivationArgs.Argument);

            // Use the originalArgs variable to access the original arguments
            // that were passed to the app.

            await Task.CompletedTask;
        }
    }
}

File: templates/Features/StoreNotificationsVB/Services/StoreNotificationsFeatureService.vb
Imports Microsoft.Services.Store.Engagement
Imports Param_RootNamespace.Activation

Namespace Services
    Friend Class StoreNotificationsFeatureService
        Inherits ActivationHandler(Of ToastNotificationActivatedEventArgs)

        Public Async Function InitializeAsync() As Task
            Dim engagementManager As StoreServicesEngagementManager = StoreServicesEngagementManager.GetDefault()
            Await engagementManager.RegisterNotificationChannelAsync()
        End Function

        Protected Overrides Async Function HandleInternalAsync(args As ToastNotificationActivatedEventArgs) As Task
            Dim toastActivationArgs = TryCast(args, ToastNotificationActivatedEventArgs)

            Dim engagementManager As StoreServicesEngagementManager = StoreServicesEngagementManager.GetDefault()
            Dim originalArgs As String = engagementManager.ParseArgumentsAndTrackAppLaunch(toastActivationArgs.Argument)

            ' Use the originalArgs variable to access the original arguments passed to the app.

            Await Task.CompletedTask
        End Function
    End Class
End Namespace 
